### PR TITLE
Add debug log statements

### PR DESCRIPTION
**Pull request by adityasaky**
This PR takes over #249.

It's possible I'm misusing the slog format here...

### CODE REVIEW

File: internal/policy/verify.go [hunk 1]
```diff
@@ -74,18 +80,21 @@ func VerifyRef(ctx context.Context, repo *git.Repository, target string) (plumbi
...
 func VerifyRefFull(ctx context.Context, repo *git.Repository, target string) (plumbing.Hash, error) {
 	// 1. Trace RSL back to the start
+	slog.Debug("Identifying first RSL entry...")
 	firstEntry, _, err := rsl.GetFirstEntry(repo)
 	if err != nil {
 		return plumbing.ZeroHash, err
 	}
 
 	// 2. Find latest entry for target
+	slog.Debug(fmt.Sprintf("Identifying latest RSL entry for '%s'...", target))
```
Comment (spectre10) [inline internal/policy/verify.go:90]: I wonder if some these generic variables like target ref for example, could be converted to key-val pairs and be used throughout the codebase. Might be helpful in filtering out logs, like for a specific ref.

Comment (wlynch) [reply internal/policy/verify.go:90]: Not a bad idea! I've been using https://pkg.go.dev/github.com/chainguard-dev/clog to plumb key/value pairs through the context.

Not blocking for this change though.

Comment (adityasaky) [reply internal/policy/verify.go:90]: @spectre10 do you want to take a stab at working on this?

Comment (spectre10) [reply internal/policy/verify.go:90]: @adityasaky I'll try `chainguard-dev/clog` and see if it's worth doing.

**Pull request merged**